[ews-build] Always clone from WebKit/WebKit

JonWBedard opened the pull request:
#### b2fe5e88d602d0a0c6fb01d7338271dfaa8c0c38
<pre>
[ews-build] Always clone from WebKit/WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=254227">https://bugs.webkit.org/show_bug.cgi?id=254227</a>
rdar://107016139

Reviewed by Aakash Jain.

We need to clone from WebKit/WebKit, even if we need to fetch
from an alternative remote.

* Tools/CISupport/ews-build/steps.py:
(CheckOutSource.__init__): Set a default repourl.
(CheckOutSource._fetch): Override the repourl with the URL of
the source repository.
(CheckOutSource.run): Move the custome repourl into _fetch.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/261976@main">https://commits.webkit.org/261976@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3488f525b21b4231e176e69bda328e70ccdb0378

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/11 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/6 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/11 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/12 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/11 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/12 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/10 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/10 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/12 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/11 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/9 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/10 "Passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/8 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/12 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -735,9 +736,18 @@ def useLog(self, *args, **kwargs):
 
-    def run(self):
+    @defer.inlineCallbacks
+    def _fetch(self, _):
```
Can we achieve same thing without overriding this internal method? 
Maybe by passing `repourl` to Git().__init__()

JonWBedard replied:
Sadly, we cannot.

The whole issue here is that we want `git fetch` to use a different URL than `git clone`. If we pass `repourl` to `Git().init()`, then if we don't have the repository on a machine yet, we'll clone from the wrong place.

aj062 reviewed (approved):
rs=me

JonWBedard committed:
[ews-build] Always clone from WebKit/WebKit
https://bugs.webkit.org/show_bug.cgi?id=254227
rdar://107016139

Reviewed by Aakash Jain.

We need to clone from WebKit/WebKit, even if we need to fetch
from an alternative remote.

* Tools/CISupport/ews-build/steps.py:
(CheckOutSource.__init__): Set a default repourl.
(CheckOutSource._fetch): Override the repourl with the URL of
the source repository.
(CheckOutSource.run): Move the custome repourl into _fetch.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/261976@main
